### ISSUE TITLE
Enhancement: Add the attribute-based parameter setters for the DbParameter object.

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#886]
### Describe the enhancement

To further support the setting of most properties of the DbParameter object before the actual execution, please add following attributes.

### Usage

```csharp
public class EntityClass
{
      [DbTypeAttribute(DbType.String)]
      [DirectionAttribute(ParameterDirection.Input)]
      [NameAttribute("CompleteName")]
      [PrecisionAttribute(1)]
      [ScaleAttribute(10)]
      [SizeAttribute(256)]
      public string Name { get; set; }
}
```

### DbParameter

Add the following attributes.

- DbTypeAttribute
- DirectionAttribute
- IsNullableAttribute
- NameAttribute
- PrecisionAttribute
- ScaleAttribute
- SizeAttribute

All the mentioned attributes must inherit from the PropertyValueAttribute attribute. 

### SqlParameter (SQL Server)

- CompareInfoAttribute
- DataTypeNameAttribute
- ForceColumnEncryptionAttribute
- LocaleIdAttribute
- OffsetAttribute
- SqlDbTypeAttribute
- XmlSchemaCollectionNameAttribute
- XmlSchemaCollectionDatabaseAttribute
- XmlSchemaCollectionOwningSchemaAttribute

### NpgsqlParameter (PostgreSQL)

- ConvertedValueAttribute
- DataTypeNameAttribute
- NpgsqlDbTypeAttribute

### MySqlParameter (MySQL)

- MySqlDbTypeAttribute

### MySqlParameter (MySQL Connector)

- MySqlDbTypeAttribute

### SQLiteParameter (System)

- TypeNameAttribute

### SqliteParameter (Microsoft)

- SqliteTypeAttribute

**Comment by mikependon** [comment]
This will be available on the next build > RepoDB v1.12.8-beta4.

**Comment by mikependon** [comment]
This is now available on version [RepoDB v1.12.8](https://www.nuget.org/packages/RepoDb/1.12.8).

**Comment by xperiandri** [comment]
@mikependon what is the purpose of `DataTypeNameAttribute` as it is never used?
